### PR TITLE
Updating processes table to contain UID and GID

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit b2ea14716c001ab8c8bedf7242e8090b616302b0 (Job results: [926](https://jenkins.osquery.io/job/osqueryPullRequestAudit/926/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit b2ea14716c001ab8c8bedf7242e8090b616302b0 (Job results: [226](https://jenkins.osquery.io/job/osqueryPullRequestOSX/226/)) failed one or more tests (**macOS/OS X**).

File: osquery/tables/system/windows/processes.cpp [hunk 1]
```diff
@@ -90,36 +99,52 @@ void genProcess(const WmiResultItem& result, QueryData& results_data) {
...
   r["resident_size"] = sPlaceHolder;
   result.GetString("VirtualSize", sPlaceHolder);
   r["total_size"] = BIGINT(sPlaceHolder);
-  results_data.push_back(r);
 
-  // OpenProcess returns nullptr on error, GetCurrentProcess returns a
-  // psuedo-handle of -1, which is the same as INVALID_HANDLE_VALUE
-  if (hProcess != nullptr && hProcess != INVALID_HANDLE_VALUE) {
-    CloseHandle(hProcess);
+  /// Get the process UID and GID from it's SID
```
Comment (theopolis) [inline osquery/tables/system/windows/processes.cpp]: Nit, its

File: osquery/tables/system/windows/processes.cpp [hunk 2]
```diff
@@ -90,36 +99,52 @@ void genProcess(const WmiResultItem& result, QueryData& results_data) {
...
+  HANDLE tok = nullptr;
+  long ret = 0;
+  unsigned long tokOwnerBuffLen;
```
Comment (theopolis) [inline osquery/tables/system/windows/processes.cpp]: Can you reduce scope?

File: osquery/tables/system/windows/processes.cpp [hunk 3]
```diff
@@ -90,36 +99,52 @@ void genProcess(const WmiResultItem& result, QueryData& results_data) {
...
+  std::vector<char> tokOwner(sizeof(TOKEN_OWNER), 0x0);
+  ret = OpenProcessToken(hProcess, TOKEN_READ, &tok);
```
Comment (theopolis) [inline osquery/tables/system/windows/processes.cpp]: Why not just auto ret =

File: osquery/tables/system/windows/processes.cpp [hunk 4]
```diff
@@ -90,36 +99,52 @@ void genProcess(const WmiResultItem& result, QueryData& results_data) {
...
+    auto sid = PTOKEN_OWNER(tokOwner.data())->Owner;
+    r["uid"] = INTEGER(getUidFromSid(sid));
+    r["gid"] = INTEGER(getGidFromSid(sid));
+  } else {
+    r["uid"] = INTEGER(uid);
+    r["gid"] = INTEGER(gid);
   }
+
+  CloseHandle(hProcess);
+  CloseHandle(tok);
```
Comment (theopolis) [inline osquery/tables/system/windows/processes.cpp]: What if tok was a nullptr?

File: osquery/core/windows/process_ops.cpp [hunk 5]
```diff
@@ -114,15 +140,15 @@ bool unsetEnvVar(const std::string& name) {
...
   buf.assign(kInitialBufferSize, '\0');
 
-  DWORD value_len =
+  auto value_len =
       ::GetEnvironmentVariableA(name.c_str(), buf.data(), kInitialBufferSize);
   if (value_len == 0) {
-    // TODO(#1991): Do we want figure out a way to be more granular in terms of
-    // the error to return?
+    VLOG(1) << "Unable to find environment variable (" << GetLastError()
+            << ") : " << name;
```
Comment (theopolis) [inline osquery/core/windows/process_ops.cpp]: Nit space between ) and :

File: osquery/core/windows/process_ops.cpp [hunk 6]
```diff
@@ -15,60 +15,61 @@
...
-  ::GetTokenInformation(token, TokenUser, nullptr, 0, &nbytes);
-  if (nbytes == 0) {
-    ::CloseHandle(token);
+  if (ret == 0) {
     return -1;
   }
+  // USER_INFO_3 struct contains the RID (uid) of our user
+  unsigned long userInfoLevel = 3;
+  unsigned char* userBuff = nullptr;
+  auto wideUserName = stringToWstring(std::string(uname.data()));
```
Comment (theopolis) [inline osquery/core/windows/process_ops.cpp]: Does this need an explicit string construction?

File: osquery/core/windows/process_ops.cpp [hunk 7]
```diff
@@ -78,30 +79,55 @@ int platformGetUid() {
...
-
   // USER_INFO_3 struct contains the RID (uid) of our user
-  DWORD userInfoLevel = 3;
-  LPUSER_INFO_3 userBuff = nullptr;
-  std::wstring wideUserName = stringToWstring(std::string(uname.data()));
-  ret = NetUserGetInfo(
-      nullptr, wideUserName.c_str(), userInfoLevel, (LPBYTE*)&userBuff);
+  unsigned long userInfoLevel = 3;
+  unsigned char* userBuff = nullptr;
+  auto wideUserName = stringToWstring(std::string(uname.data()));
```
Comment (theopolis) [inline osquery/core/windows/process_ops.cpp]: Same

File: osquery/core/windows/process_ops.cpp [hunk 8]
```diff
@@ -78,30 +79,55 @@ int platformGetUid() {
...
+  ret = NetUserGetInfo(nullptr, wideUserName.c_str(), userInfoLevel, &userBuff);
 
   if (ret != NERR_Success) {
     return -1;
   }
+  return LPUSER_INFO_3(userBuff)->usri3_primary_group_id;
```
Comment (theopolis) [inline osquery/core/windows/process_ops.cpp]: Could userBuff be a nullptr?

**Comment by osqueryer** [comment]
:-1: The commit dce3f4367aad89fe8f7e892506eb915a921ba16a (Job results: [3972](https://jenkins.osquery.io/job/osqueryPullRequestLinux/3972/)) failed one or more tests (**Linux**).

**Pull request merged**